services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java

```diff
@@ -483,4 +428,102 @@ public synchronized IpAddrRange deleteIpAddrRange(String rangeId) throws Excepti
     public synchronized IpAddrRange getIpAddrRange(String rangeId) throws Exception {
         return ipAddrRangeCache.get(rangeId);
     }
+
+    @DurationStatistics
+    public synchronized List<IpAddrAlloc> updateIpAddr(IpAddrUpdateRequest request,Map<String, List<String>> rangeToIpAddrList,Map<String, List<IpAddrRequest>> rangeRequests,
+                             Map<String, List<IpAddrRequest>> vpcIpv4Requests,Map<String, List<IpAddrRequest>> vpcIpv6Requests) throws Exception {
+        List<IpAddrAlloc> result = null;
+
+        try (Transaction tx = ipAddrRangeCache.getTransaction().start()) {
+            if (request.getOldIpAddrRequests().size() > 0) {
+                if (request.getOldIpAddrRequests().size() > 1) {
+                    releaseIpAddrBulkMethod(rangeToIpAddrList);
```

Comment:
Whether there is no need to distinguish between one or more here, just deal it as multiple case?

Reply:
Some methods require different parameters, which is complex to combine. I advised not to combine them.